### PR TITLE
Added Basic Tooltip Functionaility

### PR DESCRIPTION
**Pull request by hmtinc**
Added basic metadata parsing in the view which currently displays a basic tooltip. 

screen shot 2017-10-30 at 3 01 45 pm

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -90,6 +90,7 @@
     "chai": "^4.1.2",
     "cross-env": "^5.0.5",
     "cssnano": "^3.10.0",
+    "cytoscape-popper": "git+https://github.com/cytoscape/cytoscape.js-popper.git",
```
Comment (d2fong) [inline package.json]: Maybe it is time to release the popper extension.  @maxkfranz what do you think?

File: src/client/cytoscape-extensions.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ const compoundCollapse = require('cytoscape-compound-collapse');
 const fisheye = require('cytoscape-fisheye');
 
 //Tooltips
-// const tippy = require('cytoscape-tippy');
+const popper = require('cytoscape-popper');
```
Comment (d2fong) [inline src/client/cytoscape-extensions.js]: @hmtinc Do you know if the build file for cytoscape-popper is committed in the origin repository?

Comment (d2fong) [reply src/client/cytoscape-extensions.js]: https://github.com/cytoscape/cytoscape.js-popper/commit/b3cdcab91014b63ae2a86d794d1ac7e688a54c36
I think the build file needs to be excluded from the gitignore

File: src/client/features/view/cy/createToolTip.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
+const h = require('hyperscript');
+const classNames = require('classnames');
+const tippy = require('tippy.js');
+
+
+//Manage the creation and display of tooltips
+//Requires a valid name, cytoscape element, and parsedMetadata array
+class metadataTip {
```
Comment (d2fong) [inline src/client/features/view/cy/createToolTip.js]: classes should start with capitals e.g MetadataTip

File: testFile [hunk 4]
```diff
@@ -0,0 +1 @@
... Name","H2O"],["Comment","REPLACED http://www.ra.cs.uni-tuebingen.de/software/KEGGtranslator/cpdC00001"],["Names",[" Water","C00001"]],["Database IDs",[["kegg reaction","R02019"],["kegg reaction","R04807"],["kegg reaction","R03220"],["kegg reaction","R08941"],["kegg reaction","R03416"],["kegg reaction","R07098"],["kegg reaction","R00022"],["kegg reaction","R02720"],["kegg reaction","R08274"],["kegg reaction","R00036"],["kegg reaction","R01718"],["kegg reaction","R07052"],["kegg reaction","R00716"],["kegg reaction","R07807"],["kegg reaction","R01710"],["kegg reaction","R00122"],["kegg reaction","R05984"],["kegg reaction","R06871"],["kegg reaction","R04826"],["kegg reaction","R04204"],["kegg reaction","R04535"],["kegg reaction","R02103"],["kegg reaction","R04746"],["kegg reaction","R02092"],["kegg reaction","R02363"],["kegg reaction","R03783"],["kegg reaction","R08725"],["kegg reaction","R00335"],["kegg reaction","R00987"],["kegg reaction","R01840"],["kegg reaction","R07082"],["kegg reaction","R00286"],["kegg reaction","R03970"],["kegg reaction","R00514"],["kegg reaction","R02973"],["kegg reaction","R04738"],["kegg reaction","R04951"],["kegg reaction","R05731"],["kegg reaction","R029
```
Comment (d2fong) [inline testFile]: Is this testfile still needed?

Comment (hmtinc) [reply testFile]: No, It should have been in the .gitignore. I have removed the file now.

File: src/client/features/view/cy/createToolTip.js (hunk 3 above)
Comment (hmtinc) [reply src/client/features/view/cy/createToolTip.js]: Renamed the class and also made the file name slightly more clearer.

**Comment by d2fong** [comment]
Will do a more in depth review today

File: package.json (hunk 1 above)
Comment (maxkfranz) [reply package.json]: It still needs some work.  I haven't had a chance to review to make issues for it yet.

**Pull request merged**